### PR TITLE
Loop and include testing

### PR DESCRIPTION
**Pull request by k2tzumi**
refs https://x.com/katzchum/status/1872555008128086307

### CODE REVIEW

File: testdata/book/include_http.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      steps.get.res.status == 200
+      && steps.get.res.body.args.parameter == string(vars.parameter)
+  post:
+    desc: post
+    req:
+      /post:
+        post:
+          body:
+            application/json:
+              parameter: '{{ steps.get.res.body.args.parameter }}'
```
Comment (k2tzumi) [inline testdata/book/include_http.yml:27]: Somehow it goes to 1 on the second loop.

**Review by k2tzumi — COMMENTED** [review]
Are there race conditions?
https://github.com/k1LoW/runn/actions/runs/12542071987/job/34971386341?pr=1136

**Comment by k2tzumi** [comment]
> Are there race conditions?

It was fine.

File: testdata/book/include_http.yml (hunk 1 above)
Comment (k1LoW) [reply testdata/book/include_http.yml:27]: > Somehow it goes to 1 on the second loop.

I can't confirm the result you're talking about.
Can you still confirm it there?

<details>

```console
$ go run cmd/runn/main.go run testdata/book/loop_and_include.yml --debug
Run "loop" on "For loop & include test".steps.loop
Run "get" on "For include http test".steps.get
-----START HTTP REQUEST-----
GET /get?parameter=1 HTTP/1.1
Host: httpbin.org


-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 309
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:51 GMT
Server: gunicorn/19.9.0

{
  "args": {
    "parameter": "1"
  },
  "headers": {
    "Accept-Encoding": "gzip",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-6772428f-6cbfe5e14bd6bda63205ce38"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/get?parameter=1"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.get

Run "post" on "For include http test".steps.post
-----START HTTP REQUEST-----
POST /post HTTP/1.1
Host: httpbin.org
Content-Type: application/json

{"parameter":"1"}
-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 450
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:51 GMT
Server: gunicorn/19.9.0

{
  "args": {},
  "data": "{\"parameter\":\"1\"}",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip",
    "Content-Length": "17",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-6772428f-5b71bfed3b1e0f452d870a0a"
  },
  "json": {
    "parameter": "1"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/post"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.post
Run "get" on "For include http test".steps.get
-----START HTTP REQUEST-----
GET /get?parameter=2 HTTP/1.1
Host: httpbin.org


-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 309
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:52 GMT
Server: gunicorn/19.9.0

{
  "args": {
    "parameter": "2"
  },
  "headers": {
    "Accept-Encoding": "gzip",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724290-7fd87e3d213dafa56f35b719"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/get?parameter=2"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.get

Run "post" on "For include http test".steps.post
-----START HTTP REQUEST-----
POST /post HTTP/1.1
Host: httpbin.org
Content-Type: application/json

{"parameter":"2"}
-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 450
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:52 GMT
Server: gunicorn/19.9.0

{
  "args": {},
  "data": "{\"parameter\":\"2\"}",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip",
    "Content-Length": "17",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724290-1e05eaa82f42f63b7f56e282"
  },
  "json": {
    "parameter": "2"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/post"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.post
Run "get" on "For include http test".steps.get
-----START HTTP REQUEST-----
GET /get?parameter=3 HTTP/1.1
Host: httpbin.org


-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 309
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:54 GMT
Server: gunicorn/19.9.0

{
  "args": {
    "parameter": "3"
  },
  "headers": {
    "Accept-Encoding": "gzip",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724291-01bdb6224e726a5d541b4266"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/get?parameter=3"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.get

Run "post" on "For include http test".steps.post
-----START HTTP REQUEST-----
POST /post HTTP/1.1
Host: httpbin.org
Content-Type: application/json

{"parameter":"3"}
-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 450
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:54 GMT
Server: gunicorn/19.9.0

{
  "args": {},
  "data": "{\"parameter\":\"3\"}",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip",
    "Content-Length": "17",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724292-1eb8db232fb3637c3b063aa3"
  },
  "json": {
    "parameter": "3"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/post"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.post
Run "get" on "For include http test".steps.get
-----START HTTP REQUEST-----
GET /get?parameter=4 HTTP/1.1
Host: httpbin.org


-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 309
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:55 GMT
Server: gunicorn/19.9.0

{
  "args": {
    "parameter": "4"
  },
  "headers": {
    "Accept-Encoding": "gzip",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724293-0807057c7538327a517df383"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/get?parameter=4"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.get

Run "post" on "For include http test".steps.post
-----START HTTP REQUEST-----
POST /post HTTP/1.1
Host: httpbin.org
Content-Type: application/json

{"parameter":"4"}
-----END HTTP REQUEST-----
-----START HTTP RESPONSE-----
HTTP/2.0 200 OK
Content-Length: 450
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json
Date: Mon, 30 Dec 2024 06:49:55 GMT
Server: gunicorn/19.9.0

{
  "args": {},
  "data": "{\"parameter\":\"4\"}",
  "files": {},
  "form": {},
  "headers": {
    "Accept-Encoding": "gzip",
    "Content-Length": "17",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "Go-http-client/2.0",
    "X-Amzn-Trace-Id": "Root=1-67724293-7a7dc6ba5763e5ad18775a02"
  },
  "json": {
    "parameter": "4"
  },
  "origin": "123.198.33.120",
  "url": "https://httpbin.org/post"
}

-----END HTTP RESPONSE-----
Run "test" on "For include http test".steps.post
.

1 scenario, 0 skipped, 0 failures
$
```

</details>

Comment (k2tzumi) [reply testdata/book/include_http.yml:27]: The events are only confirmed on CI in the scenario tests that we implement in our work.
Currently, the scenario is not reproduced locally, but the sample implemented scenario is valid? is also unknown.

**Comment by k2tzumi** [comment]
The test here was a mistake, so it was unnecessary.

**Pull request closed without merging**